### PR TITLE
[GIT] add subcommands

### LINKED ISSUE #136: [GIT]: Git completion is missing --force-with-lease option.
See summary.

### PR DESCRIPTION
**Pull request by Alexanderchen929**
Exact subcommands are fetch, merge, pull, status, stage, rebase

Also closes #136

### CODE REVIEW

File: dev/git.ts [hunk 1]
```diff
@@ -1,3 +1,5 @@
+import { isOptionalChain, transpileModule } from "typescript";
```
Comment (mattschrage) [inline dev/git.ts:1]: Is this `import` intentional?

**Review by mattschrage — APPROVED** [review]
Wow! Great work - didn't realize how many options were were missing!

File: dev/git.ts (hunk 1 above)
Comment (cstrnt) [reply dev/git.ts:1]: Nope, those imports are unused. Maybe we should add `noUnusedLocals": true` to our `tsconfig.json`

File: dev/git.ts [hunk 2]
```diff
@@ -766,27 +768,348 @@ export const completionSpec: Fig.Spec = {
...
+          args: {
+            name: "mode",
+            isOptional: true,
+            suggestions: ["rebase-cousins", "no-rebase-cousins"],
+          },
+        },
+
+        {
+          name: ["-x", "--exec"],
```
Comment (cstrnt) [inline dev/git.ts:1053]: In the docs they always add "" around the commands. Should we add `insertValue: "-x "{cursor}"` here?
Screenshot 2021-05-23 at 12 31 39

File: dev/git.ts [hunk 3]
```diff
@@ -913,39 +1236,196 @@ export const completionSpec: Fig.Spec = {
       },
     },
     {
-      name: "status",
-      description: "Show the working tree status",
+      name: "stage",
+      description: "Add file contents to the staging area",
```
Comment (cstrnt) [inline dev/git.ts:1240]: Shouldn't we create a top-level const for the options and args of `git add` because they are the same?

**Review by cstrnt — CHANGES_REQUESTED** [review]
Damn, git is such a big tool 😆  I think in general we have loads of code duplication in here and maybe we should think of a way of not repeating ourselves everytime we add something new

File: dev/git.ts (hunk 1 above)
Comment (Alexanderchen929) [reply dev/git.ts:1]: agreed, think it was vscode trying to be helpful for my typos, will remove

File: dev/git.ts (hunk 2 above)
Comment (Alexanderchen929) [reply dev/git.ts:1053]: that sounds good! Thank you for this suggestion

File: dev/git.ts (hunk 3 above)
Comment (Alexanderchen929) [reply dev/git.ts:1240]: yes, was talking to brendan about this, will change around now. Thank you again!

**Comment by Ashraf-Ali-aa** [comment]
Hi, I was wondering is it possible to add the following commands.
```
	git log --color --graph --pretty=format:'%Cred%h%Creset -%C(yellow)%d%Creset %s %Cgreen(%cr) %C(bold blue)<%an>%Creset' --abbrev-commit
	git log --color --graph --pretty=format:'%Cred%h%Creset -%C(yellow)%d%Creset %s %Cgreen(%cr) %C(bold blue)<%an>%Creset' --name-status -p
```

**Review by cstrnt — APPROVED** [review]
Did some quick checks and it seems to be good! Well done, Alex 😊

**Pull request merged**